### PR TITLE
Fixes #32571 - fix reducers and fills not loading

### PR DESCRIPTION
**Pull request by MariaAga**
Using `registerReducers` in the global_index causes the components to load before the reducer, as this caused state selectors have errors because the store didn't have the plugins reducers so `state.foremanRemoteExecutionReducers` was undefined. 
Removed dup `register_global_js_file 'global'`.
Added `register_global_js_file 'fills'` to global.
Fixed functions in global_index not being functions but being called.

### CODE REVIEW

**Review by ezr-ondrej — APPROVED** [review]
I clearly see we are missing "How to plugin" guidelines :)

I like one global file and `extend` folder solution, so big :+1:
Do you think you would be able to sum it up in storybook and open up a discussion around it?

File: webpack/global_index.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import { registerRoutes } from 'foremanReact/routes/RoutingService';
 import routes from './Routes/routes';
-import registerReducers from './react_app/extend/reducers';
-import registerFills from './react_app/extend/fills';
+import fillregistrationAdvanced from './react_app/extend/fillregistrationAdvanced';
+import fillRecentJobsCard from './react_app/extend/fillRecentJobsCard';
 
 registerRoutes('foreman_remote_execution', routes);
-registerReducers();
```
[the remark this answers is not in the document]
Comment (ezr-ondrej) [reply webpack/global_index.js:7]: I'm lost again what should go in global file and what into webpack/index. And I thought I have it... apparently I don't.

[the remark this answers is not in the document]
Comment (MariaAga) [reply webpack/global_index.js:7]: Yup its fun. I'm still trying to find a good solution. The issue is if a plugin wants to use slot and fill or register routes, the reducers need to be loaded from `global_index` as  `webpack/index` isn't run in this case. But if a component is loaded using `react_component` the reducers from `global_index` will be loaded after the component, so if a user wants to load the reducer before the component they need to load it in  `webpack/index`. This causes registerReducer to run twice with the same reducer

[the remark this answers is not in the document]
Comment (ezr-ondrej) [reply webpack/global_index.js:7]: Reducers should get overriden if registered twice, right? So to be safe for now we can just register them on both the places? :thinking:

[the remark this answers is not in the document]
Comment (MariaAga) [reply webpack/global_index.js:7]: It's probably what I'll do here. But it's a bit bad to write in the docs :sweat_smile:

[the remark this answers is not in the document]
Comment (ezr-ondrej) [reply webpack/global_index.js:7]: Yeah for the docs it would be better to have a solution xD

**Comment by MariaAga** [comment]
For now I think there should be 2 `registerReducers` as it's better than having job invocation page parts break as they do now

**Review by ezr-ondrej — APPROVED** [review]
I'm good here, let's merge, just a though for the global/local I think we should make sure the globals are required before webpack/index and thus plugins can rely on it being executed always, so it can be just in the global file. But better be safe here :+1:

**Pull request merged**